src/main/java/org/openhab/binding/zigbee/converter/ZigBeeConverterColorColor.java

```diff
@@ -106,72 +266,186 @@ public void run() {
-                        level = (OnOffType) command == OnOffType.ON ? 100 : 0;
-                    } else {
-                        return;
+                        OnOffType onoff = (OnOffType) command;
+                        changeOnOff(onoff);
                     }
-
-                    clusterLevelControl.moveToLevelWithOnOffCommand((int) (level * 254.0 / 100.0 + 0.5), 10).get();
-                } catch (InterruptedException e) {
-                    e.printStackTrace();
-                } catch (ExecutionException e) {
-                    e.printStackTrace();
+                } catch ( /* InterruptedException | ExecutionException | */ Exception | Error e) {
+                    logger.warn("{}: Exception processing command", endpoint.getIeeeAddress(), e);
                 }
             }
         };
     }
 
     @Override
     public Channel getChannel(ThingUID thingUID, ZigBeeEndpoint endpoint) {
-        if (endpoint.getInputCluster(ZclColorControlCluster.CLUSTER_ID) == null
-                || endpoint.getInputCluster(ZclLevelControlCluster.CLUSTER_ID) == null) {
+        clusterColorControl = (ZclColorControlCluster) endpoint.getInputCluster(ZclColorControlCluster.CLUSTER_ID);
+
+        if (clusterColorControl == null) {
             return null;
         }
+
+        try {
+            if (!clusterColorControl.discoverAttributes(false).get()) {
+                logger.warn(
+                        "{}: Cannot determine whether device supports RGB color. Assuming it does by now (checking again later)",
+                        endpoint.getIeeeAddress());
+            } else if (!clusterColorControl.getSupportedAttributes().contains(ZclColorControlCluster.ATTR_CURRENTHUE)
+                    && !clusterColorControl.getSupportedAttributes().contains(ZclColorControlCluster.ATTR_CURRENTX)) {
+                return null;
+            }
+        } catch (InterruptedException | ExecutionException e) {
+            logger.warn(
+                    "{}: Exception checking whether device supports RGB color. Assuming it does by now (checking again later)",
+                    endpoint.getIeeeAddress(), e);
+        }
+
         return createChannel(thingUID, endpoint, ZigBeeBindingConstants.CHANNEL_COLOR_COLOR,
                 ZigBeeBindingConstants.ITEM_TYPE_COLOR, "Color");
     }
 
+    private void updateOnOff(OnOffType onoff) {
+        updateBrightness(lastBrightness);
+    }
+
+    private void updateBrightness(PercentType brightness) {
+        HSBType oldHSB = currentHSB;
+        HSBType newHSB = new HSBType(oldHSB.getHue(), oldHSB.getSaturation(), brightness);
+        currentHSB = newHSB;
+        lastBrightness = brightness;
+        updateChannelState(newHSB);
+    }
+
+    private void updateColorHSB(DecimalType hue, PercentType saturation) {
+        HSBType oldHSB = currentHSB;
+        HSBType newHSB = new HSBType(hue, saturation, oldHSB.getBrightness());
```

Comment:
I don't think we need the extra step here - I think you can just use this? -:
```
currentHSB - new HSBType(...).
```

Reply:
Same as the previous one...

Reply:
Ok